arch/arm/src/stm32h7/Kconfig

```diff
@@ -284,11 +284,55 @@ config STM32H7_HAVE_SMPS
 	bool
 	default n
 
-config STM32H7_PWR_DIRECT_SMPS_SUPPLY
-	bool "Use direct SMPS supply mode"
+choice
+	prompt "STM32 H7 Power Supply Selection"
```

Comment:
@jfbblue0922 Please build test these changes for other version of the IC. 

STM32H7_PWR_DEFAULT_SUPPLY - is not defined
STM32H7_HAVE_SMPS - will disable the choices

Reply:
I removed the 'STM32H7_PWR_DEFAULT_SUPPLY' macro in Kconfig.

I tried setting the 'STM32H7_HAVE_SMPS' macro in Kconfig to other chips.
- The "STM32 H7 Power Supply Selection" menu does not appear for STM32H7x3xx
- The "STM32 H7 Power Supply Selection" menu appears for STM32H7x5xx or STM32H7x7xx